### ISSUE TITLE
Enable more helpers

### ISSUE THREAD

**Issue by melsophos** [hans/obsidian-citation-plugin#59]
I think it could be nice to have more helpers available, for example as those [listed there](https://support.iterable.com/hc/en-us/articles/205480365-Personalizing-Templates-with-Handlebars-). The most useful would be a `replace` helper which changes a string to another. For example, I would like to replace `:` with a dot `.` instead of the default `_`. I guess that other helpers could be useful in other instance.

**Comment by coderamm** [comment]
I totally agree. Helpers would be very helpful. I would love to be able to truncate titles to a certain number of characters.

**Comment by daeh** [comment]
It would be great to be able to truncate strings on special characters like `[`, or do a regex replacement. arXiv containerTitles get inserted as `arXiv:1234.12345 [cs]`, which doesn't play nice with MD.

**Comment by alexisthual** [comment]
+1 for regex helpers!

In particular, I'd be super interested in being able to match parts of strings using regular expressions.
It would make for an easy way to automatically open zotero pdf files with [`obsidian-annotator`](https://github.com/elias-sundqvist/obsidian-annotator) as mentioned in https://github.com/hans/obsidian-citation-plugin/issues/114. For instance, one could update the Literature note template with:
```
---
...
annotator-target: {{match entry.files.[0] /\/.*\.pdf}}
---
```
where `match` would be a regex helper.

I gave a quick look at what @byteSamurai did in https://github.com/hans/obsidian-citation-plugin/pull/165, and I think it's a clean example of how we could add these helpers.

I'm happy to help adding the relevant regex helper in a PR ; tell me what you think!
